default/python/universe_generation/starsystems.py

```diff
@@ -70,6 +70,16 @@ def name_planets(system):
         fo.set_name(planet, name)
 
 
+def can_have_no_planets(star_type):
+    """
+    Return True is system is allowed to have no planets.
+    This intentionally only checks the STAR_TYPE_MOD_TO_PLANET_SIZE_DIST so that
+    the following code forces most stars to have a planetary body so that in the
+    GUI the presence of a star implies a planet
+    """
+    return universe_tables.STAR_TYPE_MOD_TO_PLANET_SIZE_DIST[star_type][fo.planetSize.noWorld] != 0
```

Comment:
Nifty, but also tricky - however, I think you should at least test for `> 0`, not `!= 0` (unless you want star types with *reduced* probability for no planet also to be classified as stars that can have no planets, which strikes me as upside down :wink:).

I admit, basing the check if a star type should allow for an empty system on the probability for no planet in that universe table has a certain appeal. However, checking for a probability `> 0` for no planet seems a bit... excessive? 

Currently, that would now include neutron stars, which have only a slightly increased probability for no planet. Which translates to "neutron stars should have less planets than normal stars" IMO, but not, like black holes, having no planets at all.

So maybe check for `> 20` or something like that?